File: Scripts/Python/Aug_03/recurrence.py
#!/usr/bin/env python3
# File: recurrence.py
# Name: D.Saravanan
# Date: 03/08/2021

""" Script for solving recurrence relation:
a(n+1) = 10 a(n) - 9 a(n-1), a(0) = 2.95, a(1) = 2.95 """

import numpy as np

N = 20
a = np.zeros(N)
a[0] = 2.95
a[1] = 2.95

for n in range(1,N-1):
    a[n+1] = 10*a[n] - 9*a[n-1]

print(a)

File: Scripts/Octave/Aug_06/recurrence.m
#!/usr/bin/env octave
% File: recurrence.m
% Name: D.Saravanan
% Date: 06/08/2021
% Script for fibonacci recurrence series

N = 20;

% exact values
e = zeros(N,1);
e(1) = 2.95;
e(2) = 2.95;

% values
a = zeros(N,1);
a(1) = 2.95;
a(2) = 2.95;

% relative error terms
err = zeros(N,1);

for n = 2:N-1
    a(n+1) = a(n) + a(n-1);
    e(n+1) = 2.95 * fibonacci(n+1);
end

for n = 1:N
    err(n) = abs((a(n)-e(n))./e(n));
end

% print relative error terms
disp(err);
